packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessage.ts

```diff
@@ -934,7 +976,15 @@ export class PaginatedMessage {
 
 		// Do not remove reactions if the message, channel, or guild, was deleted:
 		if (this.response && !PaginatedMessage.deletionStopReasons.includes(reason)) {
-			void this.response?.edit({ components: [] });
+			if (runsOnInteraction(this.response)) {
+				if (this.response.replied || this.response.deferred) {
+					void this.response?.editReply({ components: [] });
+				} else {
+					void this.response?.reply({ components: [] });
```

Comment:
This is wrong. it should call response.update

Reply:
Did you mean to comment on the editReply line instead? Surely it's not possible to call update when a message hasn't been replied to yet

Reply:
Uhh, the point is if you get a ButtonInteraction you call .update or defer it then editReply it

Reply:
This code only deals with CommandInteraction's and Select menu interaction's. Button interactions (the one from clicking the paginated buttons) are elsewhere. 

(on mobile rn otherwise I'd link you the exact lines)

Reply:
Okay i did properly check the lines. This is handleEnd which gets bound to the collector.on('end') event, so it doesn't get any specific interaction at all. What is happening here is it is cleaning up the `this.response` when the collector ends by removing the components. This code will particularly trigger when the collector times out.

Reply:
reply is still the wrong method to call